workers/loc.api/sync/win.loss/index.js

```diff
@@ -44,39 +45,95 @@ class WinLoss {
     this.authenticator = authenticator
     this.SYNC_API_METHODS = SYNC_API_METHODS
 
+    this.positionsHistoryModel = this.syncSchema.getModelsMap()
+      .get(this.ALLOWED_COLLS.POSITIONS_HISTORY)
     this.movementsModel = this.syncSchema.getModelsMap()
       .get(this.ALLOWED_COLLS.MOVEMENTS)
     this.movementsMethodColl = this.syncSchema.getMethodCollMap()
       .get(this.SYNC_API_METHODS.MOVEMENTS)
     this.movementsSymbolFieldName = this.movementsMethodColl.symbolFieldName
+    this.positionsSnapshotMethodColl = this.syncSchema.getMethodCollMap()
+      .get(this.SYNC_API_METHODS.POSITIONS_SNAPSHOT)
+    this.positionsSnapshotSymbolFieldName = this.positionsSnapshotMethodColl.symbolFieldName
   }
 
-  async _getPlFromPositionsSnapshot (args) {
-    const positionsSnapshot = await this.positionsSnapshot
-      .getSyncedPositionsSnapshot(args)
+  _isClosedPosition (positionsHistory, mts, id) {
+    return (
+      Array.isArray(positionsHistory) &&
+      positionsHistory.length > 0 &&
+      positionsHistory.some((item) => (
+        item.id === id &&
+        item.mts === mts
+      ))
+    )
+  }
 
+  _filterPositionsSnapshots (
+    positionsSnapshots,
+    positionsHistory,
+    mts
+  ) {
     if (
-      !Array.isArray(positionsSnapshot) ||
-      positionsSnapshot.length === 0
+      !Array.isArray(positionsSnapshots) ||
+      positionsSnapshots.length === 0
     ) {
-      return null
+      return positionsSnapshots
     }
 
-    return positionsSnapshot.reduce((accum, curr) => {
-      const { plUsd } = { ...curr }
-      const symb = 'USD'
+    const orderedPositions = orderBy(
+      positionsSnapshots,
+      ['mtsUpdate', 'id'],
+      ['desc', 'desc']
+    )
 
-      if (!Number.isFinite(plUsd)) {
-        return accum
+    return orderedPositions.reduce((accum, position) => {
+      if (
+        position &&
+        typeof position === 'object' &&
+        accum.every((item) => item.id !== position.id) &&
+        !this._isClosedPosition(positionsHistory, mts, position.id)
+      ) {
+        accum.push(position)
       }
 
-      return {
-        ...accum,
-        [symb]: Number.isFinite(accum[symb])
-          ? accum[symb] + plUsd
-          : plUsd
-      }
-    }, {})
+      return accum
+    }, [])
+  }
+
+  _calcPlFromPositionsSnapshots (positionsHistory) {
+    return (
+      data = [],
+      args = {}
+    ) => {
+      const { mts, timeframe } = args
+
+      // Need to filter duplicate and closed positions as it can be for
+      // week and month and year timeframe in daily positions snapshots
+      // if daily timeframe no need to filter it
+      const positions = timeframe !== 'day'
+        ? this._filterPositionsSnapshots(
+            data,
+            positionsHistory,
+            mts
+          )
+        : data
+
+      return positions.reduce((accum, curr) => {
```

Comment:
Not sure if this would be accurate.
The idea is to, just the positions that where open at the start or end check the pl of the start or end. 
Have into consideration that as to consider the pl of the end just one position snapshot is required per trading pair. 
Same for start. 
So would be one for the start and one for the end when corresponds.